lib/galaxy/tools/parameters/basic.py

```diff
@@ -1967,18 +1965,22 @@ def single_to_python(value):
         if value in none_values:
             return None
         if isinstance(value, str) and value.find(",") > -1:
-            return [session.get(HistoryDatasetAssociation, int(v)) for v in value.split(",") if v not in none_values]
+            return [
+                app.model.context.get(HistoryDatasetAssociation, int(v))
```

Comment:
For the future, my suggestion would be to use a local `session` variable instead of `app.model.context`: it makes our (often quite heavy) data access code more readable, and we won't be tied to that chain of indirection (`app` must have `model`, `model` must have `context`). Besides, we use `session` or `sa_session` all over the code base, so it's recognizable and consistent.